components/scream/src/physics/share/tests/physics_universal_unit_tests.cpp

```diff
@@ -0,0 +1,205 @@
+    Real expected_T = T_in*exner[0];
+    if (std::abs(T_atm[0]-expected_T)>tol) {
+      printf("T to th test: abs(th_atm-expected_th)=%e is larger than the tol=%e\n",std::abs(T_atm[0]-expected_T),tol);
+      errors++;
+    }
+
+    // Test that T_to_th and th_to_T are inverses of each other.
+    // Test T to th as inverses of each other
+    // Converting T to th and then back to T should return the same value again.
+    const Spack th_temp = physics::T_to_th(T,exner,Smask(true));
+    const Spack T_new   = physics::th_to_T(th_temp,exner,Smask(true));
+    if (std::abs(T_new[0]-T_in)>tol) {
+      printf("T to th test: abs[ T_new (%.3e) - T_in (%.3e) ]=%e is larger than the tol=%e\n",T_new,T_in,std::abs(T_new[0]-T_in),tol);
+      errors++;
+    }
+    // and vice versa
+    const Spack T_temp = physics::th_to_T(T,exner,Smask(true));
+    const Spack th_new = physics::T_to_th(T_temp,exner,Smask(true));
+    if (std::abs(T_new[0]-T_in)>tol) {
+      printf("T to th test: abs[ th_new (%.3e) - T_in (%.3e) ]=%e is larger than the tol=%e\n",th_new,T_in,std::abs(th_new[0]-T_in),tol);
+      errors++;
+    }
+  } // T_th_conversion_test
+//-----------------------------------------------------------------------------------------------//
+  KOKKOS_FUNCTION static void dz_tests(const Scalar& zi_top, const Scalar& zi_bot, int& errors){
+
+    // Allow usage of universal functions
+    using physics = scream::physics::Functions<Scalar, Device>;
+    // Gather the test tolerance
+    static constexpr Scalar eps = C::macheps;
+    Real tol = 1000*eps;
+
+    //========================================================
+    // Test calculation of layer thickness using get_dz 
+    //========================================================
+    // This function tests the function "get_dz" for a set of interface layer heights.
+    //
+    // Inputs:
+    //   zi_top: The above surface height of the top of the verical layer, m.
+    //   zi_bot: The above surface height of the bottom of the verical layer, m.
+    // Outputs:
+    //   errors:  A tally of any errors that this test detects.
+    //========================================================
+
+    const Spack z_top(zi_top);
+    const Spack z_bot(zi_bot);
+    Real expected_dz = zi_top-zi_bot;
+    const Spack dz = physics::get_dz(z_top, z_bot, Smask(true));
+    if (std::abs(dz[0]-expected_dz)>tol) {
+      printf("get_dz test: abs(dz-expected_dz)=%e is larger than the tol=%e\n",std::abs(dz[0]-expected_dz),tol);
+      errors++;
+    }
+
+  } // dz_test 
+
+//-----------------------------------------------------------------------------------------------//
+  static void run()
+  {
+    using physics = scream::physics::Functions<Scalar, Device>;
+
+    int nerr = 0;
+    TeamPolicy policy(ekat::ExeSpaceUtils<ExeSpace>::get_default_team_policy(1, 1));
+    Kokkos::parallel_reduce("test_universal_physics", policy, KOKKOS_LAMBDA(const MemberType&, int& errors) {
+
+      errors = 0;
```

Comment:
Since this is a sum reduction, the error count should be accumulated, i.e., `error += ...`. Setting it to 0 has undefined and very likely incorrect results.

Reply:
oh yes, good catch.  I'll make this change and push a the fix.